maya/mel/assign-pbr-textures.mel

```diff
@@ -0,0 +1,43 @@
+{
+    $sl = {""};
+    $sl = `ls -sl`;
+    
+    if($sl[0]=="")
+    {
+        print "No selection to apply the material to...\n";
+    }
+    else
+    {
+        loadPlugin "C:/Program Files/Autodesk/Maya2017/bin/plug-ins/glslShader.mll";
+        source "createAndAssignShader.mel";
+    
+        $sh = `shadingNode -asShader GLSLShader`;
+        setAttr -type "string" ($sh+".shader") "D:\\Development\\PBR\\shaders\\glTF_PBR.ogsfx";
+        assignCreatedShader "GLSLShader" "" $sh $sl[0];
+        
+        $numAttrs = 9;
+        $attrNames = {
+            "u_BaseColorTexture",
+            "u_MetallicTexture",
+            "u_RoughnessTexture",
+            "u_OcclusionTexture",
+            "u_NormalTexture",
+            "u_EmissiveTexture",
+            "u_DiffuseEnvTexture",
+            "u_SpecularEnvTexture",
+            "u_brdfTexture"
+        };
+        
+        for ($i=0;$i<$numAttrs;++$i)
+        {
+            $shAttrName = ($sh+"."+$attrNames[$i]);
+            $filePath = {""};
+            $filePath = `fileDialog2 -cap ("Open '"+$shAttrName+"'") -fm 1`;
```

Comment:
Asking for every filename is one approach. But artists will still complain about  the many dialog boxes. Another common approach in studios might be naming conventions. I would add a parameter to the procedure with these naming conventions (e.g. an array of regular expressions for each of the texture attributes). If this array is passed, then let the user select many textures files at once, and then test each file again the reg-ex, putting it in the right file node, if that reg-ex matches. Otherwise do not create a file node at all. What do you think?